cocos/serialization/deserialize.ts

```diff
@@ -1018,54 +1029,47 @@ export function isCompiledJson (json: unknown): boolean {
  * @param options Deserialization Options.
  * @return The original object.
  */
-export function deserialize (data: IFileData | string | CCON | any, details: Details | any, options?: IOptions | any): unknown {
+export function deserialize (data: IDeserializeInput | string | any, details?: Details, options?: IOptions & DeserializeDynamicOptions): unknown {
     if (typeof data === 'string') {
         data = JSON.parse(data);
     }
 
-    const borrowDetails = !details;
-    details = details || Details.pool.get();
+    let isBorrowedDetails = false;
+    if (!details) {
+        const borrowedDetails = Details.pool.get();
```

Comment:
`Details.pool.get()` gives `Details | null`, whereas `details` is `Details | undefined`. Otherwise here should be write as:

```diff
  let isBorrowedDetails = false;
  if (!details) {
-    const borrowedDetails = Details.pool.get();
-    assertIsTrue(borrowedDetails , `Can not allocate deserialization details`);
-    details = borrowedDetails ;
+    details = Details.pool.get() ?? undefined;
+    assertIsTrue(details, `Can not allocate deserialization details`);
      isBorrowedDetails = true;
}
```